src/main/kotlin/com/rose/gateway/minecraft/logging/Logger.kt

```diff
@@ -3,13 +3,22 @@ package com.rose.gateway.minecraft.logging
 import com.rose.gateway.GatewayPlugin
 import org.koin.core.component.KoinComponent
 import org.koin.core.component.inject
+import java.util.logging.FileHandler
 
 /**
  * Helper object that provides logging functionality throughout Gateway
  */
 object Logger : KoinComponent {
     private val plugin: GatewayPlugin by inject()
     private val pluginLogger = plugin.logger
+    private val handler: FileHandler
+
+    init {
+        handler = FileHandler(plugin.dataFolder.path.plus("/gateway_log.txt"), true)
+        pluginLogger.addHandler(handler)
+        val formatter = LogFormatter()
+        handler.formatter = formatter
```

Comment:
You could just combine the instantiation of LogFormatter to the setting of the handler formatter. The variable is unnecessary.